flybywiresim/aircraft
# feat(RMP): implement LOAD logic in combination with CPDLC

## Summary of Changes
- Extend the ATSU to send Arinc429Register to RMP
- Send the frequency at "CONTACT XXX" or "MONITOR XXX" messages
- Add emissive code to control the LED

## Screenshots (if necessary)


![{0DAC8B02-C1CB-4CE2-AB62-6E5D1ED0D92D}](https://github.com/user-attachments/assets/2a2c2882-412d-4bbd-a9ff-7214da92587f)

## References

Feedback from pilots and FCOM




## Additional context



Discord username (if different from GitHub):

## Testing instructions

- Connect to Hoppie ACARS (also with the ATC software)
- Send "CONTACT XXX", "AT XXX CONTACT YYY", "MONITOR XXX", "AT XXX MONITOR YYY" to the aircraft
- Message received and visible on DCDU
  - LOAD LED on RMP should enlight
  - Load the frequency (to standby frequency)
  - LED should be dark again
- Message not visible on DCDU
  - LED should be dark
  - Click on LOAD changes nothing
- Message visible on DCDU as RECALL
  - LED should be dark
  - Click on LOAD changes nothing


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
